agent/xds/server.go

```diff
@@ -328,6 +328,10 @@ func (s *Server) process(stream ADSStream, reqCh <-chan *envoy.DiscoveryRequest)
 			//  2. Non-determinsic order of complex protobuf responses which are
 			//     compared for non-exact JSON equivalence makes the tests uber-messy
 			//     to handle
+
+			// TODO(rb): maintain old and new discovery chains while we are transitioning
+			// TODO(rb): the clusters that we will be using to avoid dropping traffic
+			// TODO(rb): we likely have to do our own fascinating blue/green setup
```

Comment:
Is this still TODO? I think you mentioned figuring out a magic setting to avoid dropping traffic?

Reply:
Nope. Outdated comment. Ended up being the `validate_clusters` envoy setting having different defaults a route came in statically vs dynamically: https://www.envoyproxy.io/docs/envoy/latest/api-v2/api/v2/rds.proto.html#envoy-api-field-routeconfiguration-validate-clusters